File: gda-java-components/src/main/java/programmingtheiot/data/SystemPerformanceData.java
package programmingtheiot.data;

import java.io.Serializable;

import programmingtheiot.common.ConfigConst;

public class SystemPerformanceData extends BaseIotData implements Serializable
{
	private static final long serialVersionUID = 1L;

	private float cpuUtil  = ConfigConst.DEFAULT_VAL;
	private float diskUtil = ConfigConst.DEFAULT_VAL;
	private float memUtil  = ConfigConst.DEFAULT_VAL;

	public SystemPerformanceData()
{
	super();

	super.setName(ConfigConst.SYS_PERF_DATA);
	super.setTypeID(ConfigConst.SYSTEM_PERF_TYPE);
}
	public float getCpuUtilization()
	{
		return this.cpuUtil;
	}

	public float getDiskUtilization()
	{
		return this.diskUtil;
	}

	public float getMemoryUtilization()
	{
		return this.memUtil;
	}

	public void setCpuUtilization(float cpuUtil)
	{
		updateTimeStamp();
		this.cpuUtil = cpuUtil;
	}

	public void setDiskUtilization(float diskUtil)
	{
		updateTimeStamp();
		this.diskUtil = diskUtil;
	}

	public void setMemoryUtilization(float memUtil)
	{
		updateTimeStamp();
		this.memUtil = memUtil;
	}

	@Override
	protected void handleUpdateData(BaseIotData data)
	{
		if (data instanceof SystemPerformanceData) {
			SystemPerformanceData spData = (SystemPerformanceData) data;

			this.setCpuUtilization(spData.getCpuUtilization());
			this.setDiskUtilization(spData.getDiskUtilization());
			this.setMemoryUtilization(spData.getMemoryUtilization());
		}
	}
}

File: cda-python-components/programmingtheiot/data/SystemPerformanceData.py
import programmingtheiot.common.ConfigConst as ConfigConst
from programmingtheiot.data.BaseIotData import BaseIotData


class SystemPerformanceData(BaseIotData):
    """
    Data container for system performance telemetry.
    """

    def __init__(self, d=None):
        super(SystemPerformanceData, self).__init__(
            name=ConfigConst.SYSTEM_PERF_MSG,
            typeID=ConfigConst.SYSTEM_PERF_TYPE,
            d=d
        )

        self.cpuUtil = ConfigConst.DEFAULT_VAL
        self.memUtil = ConfigConst.DEFAULT_VAL

    def getCpuUtilization(self):
        return self.cpuUtil

    def getMemoryUtilization(self):
        return self.memUtil

    def setCpuUtilization(self, cpuUtil):
        self.cpuUtil = cpuUtil
        self.updateTimeStamp()

    def setMemoryUtilization(self, memUtil):
        self.memUtil = memUtil
        self.updateTimeStamp()

    def _handleUpdateData(self, data):
        try:
            if data and isinstance(data, SystemPerformanceData):
                self.cpuUtil = data.getCpuUtilization()
                self.memUtil = data.getMemoryUtilization()
        except Exception:
            pass

    def __str__(self):
        return "SystemPerformanceData [cpuUtil={}, memUtil={}, timeStamp={}]".format(
            self.cpuUtil, self.memUtil, self.getTimeStamp()
        )